MapboxVisionTests/RecordingQuotaTests.swift

```diff
@@ -0,0 +1,124 @@
+import XCTest
+@testable import MapboxVision
+
+typealias Byte = Int64
+
+private let bytesInKByte: Byte = 1024
+private let kByteInMByte: Byte = 1024
+
+private let kByte = bytesInKByte
+private let mByte = kByteInMByte * kByte
+
+private let secondsInMinute: TimeInterval = 60
+private let minutesInHour: TimeInterval = 60
+
+private let minute = secondsInMinute
+private let hour = secondsInMinute * minute
+
+private let memoryLimit = 300.0 // mb
+private let networkingMemoryLimit = 30 * mByte
+private let updatingInterval = 1 * hour
+
+class RecordingQuotaTests: XCTestCase {
+    // MARK: - Properties
+
+    private var refreshInterval: TimeInterval!
+    private var initialMemoryQuota: Byte!
+    private var recordingQuota: RecordingQuota!
+
+    // MARK: - Test functions
+
+    override func setUp() {
+        super.setUp()
+        refreshInterval = TimeInterval.random(in: 5...10)
```

Comment:
These time intervals can drastically increase the time it takes to run tests.
We can discuss how to avoid this, but one way is to pass current date as a parameter to `reserve(memoryToReserve:)` method with a default value. WDYT?

Reply:
To be honest I don't like the idea with the additional date parameter much. I've decreased a maximum delay, in total right now it's up to 10 seconds. Do you agree to leave it in that form?